sdk/eventgrid/azure-eventgrid/README.md

```diff
@@ -38,26 +38,37 @@ In order to interact with the Event Grid service, you will need to create an ins
 A **topic_hostname** and **credential** are necessary to instantiate the client object.
 
 #### Looking up the endpoint
-You can find the endpoint and the hostname on the Azure portal.
+You can find the topic endpoint within the Event Grid Topic resource on the Azure portal.
+The topic hostname is the URL host component of this endpoint. (Everything up-to and including "eventgrid.azure.net".)
 
 #### Create the client with AzureKeyCredential
 
-To use an API key as the `credential` parameter,
+To use an Access Key as the `credential` parameter,
 pass the key as a string into an instance of [AzureKeyCredential][azure-key-credential].
 
+> **Note:** The Access Key may be found in the azure portal in the "Access Keys" menu of the Event Grid Topic resource.  They may also be obtained via the azure CLI, or the `azure-mgmt-eventgrid` library.
+
 ```python
 from azure.core.credentials import AzureKeyCredential
 from azure.eventgrid import EventGridPublisherClient
 
 topic_hostname = "https://<name>.<region>.eventgrid.azure.net"
-credential = AzureKeyCredential("<api_key>")
+credential = AzureKeyCredential("<access_key>")
 eg_publisher_client = EventGridPublisherClient(topic_hostname, credential)
 ```
 
+> **Note:** A client may also be authenticated via SAS signature, using the `EventGridSharedAccessSignatureCredential`.  A sample demonstrating this, as well as how to generate that signature utilizing `generate_shared_access_signature` is available [here][python-eg-sample-publish-sas-signature].
+
 ## Key concepts
 
 Information about the key concepts on Event Grid, see [Concepts in Azure Event Grid][publisher-service-doc]
 
+### Topic
+A channel within the EventGrid service to send events.  Must be of CloudEvent or EventGridEvent schema, (decided at creation time) and the corrosponding event type must be used.
+
+### Domain Topic
+A domain exists to rout arbitrary named topics within it.  Topics must not exist beforehand if using a domain, topic name can be provided at send time.  Simply provide the domain endpoint at client construction instead of topic endpoint.
```

Comment:
A little unclear on what is meant by `Topics must not exist beforehand if using a domain, topic name can be provided at send time.`? I tried creating a domain and publishing to the domain endpoint with the topic name specified, but the topic did not exist beforehand. It was published and reached the endpoint, but the event was not delivered to that topic. I had to manually name and create the domain topic in the portal (or management plane sdk) under the domain, and published again before it was delivered there.

Might be worthwhile to make it more clear how a domain topic works in comparison to other topics + to include things from this doc (https://docs.microsoft.com/en-us/azure/event-grid/event-domains) including the following:
1) More info on what a domain is: "An event domain is a management tool for large numbers of Event Grid topics related to the same application. You can think of it as a meta-topic that can have thousands of individual topics." The way it's phrased now doesn't really show why domains are used, how the individual topics are related, etc.
2) Publishing to a domain: "When you create an event domain, you're given a publishing endpoint similar to if you had created a topic in Event Grid...
To publish events to any topic in an Event Domain, push the events to the domain's endpoint the same way you would for a custom topic. The only difference is that you must specify the topic you'd like the event to be delivered to."

 Links + typo + something like this?
```suggestion
A domain is a management tool that exists to route events to individual domain topics within it which are related to the same application. Publishing to a domain topic is similar to publishing to a custom topic. There are two key differences:
1. At client construction, the domain endpoint must be provided instead of topic endpoint. 
2. The individual domain topic name must be specified in the event at send time. To be delivered to the domain topic provided, the domain topic must already exist in the domain.

More information about Event Domains can be found [here](https://docs.microsoft.com/en-us/azure/event-grid/event-domains).
```

Reply:
as @swathipil mentioned, echoing words from the service doc is good.

also I'm feeling that as it is the `Key concepts` section, `Domain topic` could just describe the service concepts and we leave the details of client constructor ("only domain endpoint at client construction for domain topic..." etc.) could be placed under the `EventGridPublisherClient` section.